lib/devicetrust/native/api.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package native
+
+import devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+
+// EnrollDeviceInit creates the initial enrollment data for the device.
+// This includes fetching or creating a device credential, collecting device
+// data and filling in any OS-specific fields.
+func EnrollDeviceInit() (*devicepb.EnrollDeviceInit, error) {
```

Comment:
Why not have these just implemented as part of the correct native package instead? What's the trampoline for?

Reply:
The trampoline helps me ensure all build tags are equivalent. It's also an attempt at making the APIs easier to visualize, by having them explained/documented in a single file. (Also godocs are only written once.)